docs/content/integrations/duckdb/using-duckdb-with-dagster.mdx

```diff
@@ -30,45 +29,203 @@ To complete this tutorial, you'll need:
+from dagster import asset
+
+# this example uses the iris_dataset asset from Step 1
+
+
+@asset(deps=[iris_dataset])
+def iris_setosa(duckdb: DuckDBResource) -> None:
+    with duckdb.get_connection() as conn:
+        conn.execute(
+            "CREATE TABLE iris.iris_setosa AS SELECT * FROM iris.iris_dataset WHERE"
+            " species = 'Iris-setosa'"
+        )
+```
+
+In this asset, you create second table that just contains the data for the Iris Setosa species. This asset has a dependency on the `iris_dataset` asset. To define this dependency, you provide the `iris_dataset` asset as a `dep` to the `iris_setosa` asset. You can then run the SQL query to create the table of Iris Setosa data.
+
+### Completed code example
+
+When finished, your code should look like the following:
+
+```python file=/integrations/duckdb/tutorial/resource/full_example.py
+import pandas as pd
+from dagster_duckdb import DuckDBResource
+
+from dagster import Definitions, SourceAsset, asset
+
+iris_harvest_data = SourceAsset(key="iris_harvest_data")
+
+
+@asset
+def iris_dataset(duckdb: DuckDBResource) -> None:
+    iris_df = pd.read_csv(
+        "https://docs.dagster.io/assets/iris.csv",
+        names=[
+            "sepal_length_cm",
+            "sepal_width_cm",
+            "petal_length_cm",
+            "petal_width_cm",
+            "species",
+        ],
+    )
+
+    with duckdb.get_connection() as conn:
+        conn.execute("CREATE TABLE iris.iris_dataset AS SELECT * FROM iris_df")
+
+
+@asset(deps=[iris_dataset])
+def iris_setosa(duckdb: DuckDBResource) -> None:
+    with duckdb.get_connection() as conn:
+        conn.execute(
+            "CREATE TABLE iris.iris_setosa AS SELECT * FROM iris.iris_dataset WHERE"
+            " species = 'Iris-setosa'"
+        )
+
+
+defs = Definitions(
+    assets=[iris_dataset],
+    resources={
+        "duckdb": DuckDBResource(
+            database="path/to/my_duckdb_database.duckdb",
+        )
+    },
+)
+```
+
+## Part 2: Using the DuckDB I/O Manager
+
+You may want to use an I/O manager to handle storing DataFrames as tables in DuckDB and loading DuckDB tables as DataFrames in downstream assets. Using an I/O manager is not required, and you can reference [When to use I/O managers](TODO-LINK) to learn more.
+
+This section of the guide focuses on storing and loading Pandas DataFrames in DuckDB, but Dagster also supports using PySpark and Polars DataFrames with DuckDB. The concepts from this guide apply to working with PySpark and Polars DataFrames, and you can learn more about setting up and using the DuckDB I/O manager with PySpark and Polars DataFrames in the [reference guide](/integrations/duckdb/reference).
+
+### Step 1: Configure the DuckDB I/O manager
+
+The DuckDB I/O manager requires some configuration to connect to your database. You must provide a path where a DuckDB database will be created. Additionally, you can specify a `schema` where the DuckDB I/O manager will create tables.
```

Comment:
It feels nitpicky, but this section doesn't talk about `Definitions` at all while the Step 1 of Part 1 does. If we're telling folks they can choose one approach or the other (or both), perhaps these similar sections should be similarly specific.

Reply:
good point